smarty/sampler.py

```diff
@@ -129,41 +144,16 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
                 self.atomtypes = [(smarts, typename)] + self.atomtypes
```

Comment:
@bannanc wants to have the basetypes in order, the most general to the least general. If WE don't do that by hand on the basetype file, I don't know how we can treat them.
When the basetype filename has the basetypes in order, we don't have the problem of [$ewg1] showing before than [*].

> Also, "That should fix this" is unclear to me -- what's the "that" and the "this" that are referred to?

That: "Shouldn't these be listed as most general to least general? so the [*] option would come first."
This: "So I think what should happen is that you add any "missing" basetypes to the top of the initial atomtype list in the same order as they appeared in your basetype list."

Again, if the basetype file is in order, any "missing" basetypes will show to the top of the initial atomtype list.

**This is how the algorithm is doing:**
```
USED BASE TYPES: [('[*]', 'any_atom'), ('[$ewg1]', 'ewg1'), ('[$ewg2]', 'ewg2'), ('[#1]', 'hydrogen'), ('[#6]', 'carbon'), ('[#8]', 'oxygen')]

Added base (generic) type `[*]`, name c_any_atom, to initial types.
Added base (generic) type `[$ewg1]`, name c_ewg1, to initial types.
Added base (generic) type `[$ewg2]`, name c_ewg2, to initial types.
Added base (generic) type `[#1]`, name c_hydrogen, to initial types.
Added base (generic) type `[#6]`, name c_carbon, to initial types.
Added base (generic) type `[#8]`, name c_oxygen, to initial types.

INDEX        ATOMS  MOLECULES                                                          TYPE NAME                           SMARTS
    1 :          0          0 |                                                       c_any_atom                              [*]
    2 :          0          0 |                                                           c_ewg1                          [$ewg1]
    3 :          0          0 |                                                           c_ewg2                          [$ewg2]
    4 :        464         42 |                                                       c_hydrogen                             [#1]
    5 :        232         42 |                                                         c_carbon                             [#6]
    6 :          0          0 |                                                       c_nitrogen                             [#7]
    7 :        107         42 |                                                         c_oxygen                             [#8]
    8 :          0          0 |                                                       c_fluorine                             [#9]
    9 :          0          0 |                                                    c_phosphorous                            [#15]
   10 :          0          0 |                                                         c_sulfur                            [#16]
   11 :          0          0 |                                                       c_chlorine                            [#17]
   12 :          0          0 |                                                       c_selenium                            [#34]
   13 :          0          0 |                                                        c_bromine                            [#35]
   14 :          0          0 |                                                         c_iodine                            [#53]
TOTAL :        803         42
```